src/Blockcore.Indexer.Core/Sync/SyncTasks/HistoryComputer.cs

```diff
@@ -0,0 +1,62 @@
+using System.Threading.Tasks;
+using Blockcore.Indexer.Core.Extensions;
+using Blockcore.Indexer.Core.Settings;
+using Blockcore.Indexer.Core.Storage;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Blockcore.Indexer.Core.Sync.SyncTasks;
+
+public class HistoryComputer : TaskRunner
+{
+   readonly ILogger<HistoryComputer> logger;
+   readonly IStorage storage;
+   readonly IComputeHistoryQueue computeHistoryQueue;
+
+   public HistoryComputer(IOptions<IndexerSettings> configuration, ILogger<HistoryComputer> logger, IStorage storage,
+      IComputeHistoryQueue computeHistoryQueue) : base(configuration, logger)
+   {
+      this.logger = logger;
+      this.storage = storage;
+      this.computeHistoryQueue = computeHistoryQueue;
+   }
+
+   public override async Task<bool> OnExecute()
+   {
+      if (StopRunnerExecution)
+      {
+         return false;
+      }
+
+      var stopwatch = Stopwatch.Start();
+
+      int addressesCount = 0;
+
+      do
+      {
+         if (!computeHistoryQueue.GetNextItemFromQueue(out string address))
+         {
+            stopwatch.Stop();
+            return await Task.FromResult(false);
+         }
+
+         storage.AddressBalance(address);
```

Comment:
As discussed we should log on each loop and we should also log the backlog so we have an idea of how many items are waiting in the queue

Reply:
Since we limit the number of items in the queue we won't need to log it every time